build/bin.NOTICE

```diff
@@ -165,7 +181,7 @@ ip4s (lib/com.comcast:ip4s-core.jar-<VERSION>.jar)
 
 Apache Commons Lang (lib/org.apache.commons:commons-lang3-<VERSION>.jar)
   Apache Commons Lang
-  Copyright 2001-2021 The Apache Software Foundation
-  
+  Copyright 2001-2022 The Apache Software Foundation
+
   This product includes software developed at
   The Apache Software Foundation (https://www.apache.org/).

```

Comment:
We just made a bunch of corrections to this file that are accurate for 3.3.0. You might need to incorporate some of those changes.

Reply:
@stevedlawrence So I updated this on the weekend to contain the contents I found in `daffodil/NOTICE` was that not the correct fix for this?

Reply:
That was correct, but @tuxji opened a new PR (https://github.com/apache/daffodil/pull/777) to fix some NOTICE issues we found during the Daffodil 3.3.0 vote. We decided it wasn't worth blocking the release and fixed them shortly after. That PR hasn't been merged yet, but should probably be merged relatively soon. At which point the relevant parts can be copied in. Note that PR also makes a minor update to the license file too.

Reply:
So I am confused. That PR is for updating the `bin.NOTICE` and `bin.LICENSE` files for Daffodil. These wouldn't be inside of the `bin.NOTICE` and `bin.LICENSE` correct? Just the root `NOTICE` and `LICENSE` file contents should be inside of the `bin.` files inside of `daffodil-vscode` correct? As the `bin` files between the two should differ as the daffodil-debugger Scala code doesn't use the same dependencies as the daffodil project.

Reply:
Okay I can work on getting these changes updated, should I just the contents of that PR or whats in main?

Reply:
You might wait until things are merged. I'm not sure what the final PR will look like, and the more similar the two projects are, the easier it is to verify.

Reply:
Okay that makes sense, do you want those other dependencies removed or anything else updated before I merge in this PR in? After merge would I be good to release RC3 or do you think I should wait till that daffodil PR is merged than fix the bin files and make another PR or keep this one open till that time?

Reply:
I think the dependency changes look fine. RC3 should be good once this is merged, but I might make sense to include the bin.NOTICE and bin.LICENSE updates in this PR. Also, did you confirm that the updated dependencies did not have a change to their LICENSE or NOTICE files that needs to be incorporated into bin.NOTICE/bin.LICENE?

Reply:
Yeah for all the ones I could fine most didn't change or if they did it was normally a simple bump of copyright year. The main thing was not knowing what I needed to copy over from daffodil files.

Reply:
@stevedlawrence Hey Steve, I am going through and updating stuff here so if something doesn't have a NOTICE but has a LICENSE it can be excluded from `bin.NOTICE` if the license is added to `bin.LICENSE` correct? What about something that has neither LICENSE or NOTICE does that just not get added to either file? What if the LICENSE is an Apache license does that need added to that file or we bypass those?

Reply:
So we have these items included but they aren't in either `bin.NOTICE` or `bin.LICENSE` but have Apache Licenses
```
org.typelevel.cats-effect
org.typelevel.cats-effect-kernel
org.typelevel.cats-effect-std
com.monovore.decline-effect
com.monovore.decline
org.typelevel.log4cats-core
org.typelevel.log4cats-slf4j
com.google.code.gson.gson
org.typelevel.literally
```
This one is in neither and has neither
```
org.typelevel.simulacrum-scalafix-annotations
```
So what do you think is best for these ones just leave them out? Adding the Apache license for all of the above, some are in the same project so it lowers the number down a bit, will cause that files to a bit more lengthy

Reply:
For dependencies that are under the Apache License, you must add the dependencies NOTICE content to bin.NOTICE. If it does not have a NOTICE file then you do not need to do anything for that dependency. And because it's ALv2 license, you not need to add anything to the bin.LICENSE file since it's the same as the vscode license.

For dependencies that are under a license other than the Apache License, you must add its LICENSE information to the bin.LICENSE file. Nothing should be added to the bin.NOTICE file, unless the license is a [Category B](https://www.apache.org/legal/resolved.html#category-b) license, but we don't see those very often, and in most cases the information should be in a README rather than the bin.NOTICE.

If a dependency does not have a LICENSE file, then we need to figure out what its license is and handle it according to the above rules. For example, sometimes the license is just mentioned in a README or file headers. If we cannot determine the license, it is assumed to be [Category X](https://www.apache.org/legal/resolved.html#category-x) and we cannot use the dependency until the author/authors specify a license and we determine its compatibility with ASF licensing.

Reply:
Sorry, I didn't see the other comments until after I posted the above, but I think it answers your questions.

> Also in the PR for daffodil some of the items that have no NOTICE in JAR have the license mentioned in bin.LICENSE so could those just be removed from the bin.NOTICE? Some of them also have Apache licenses so should those be removed from the bin.NOTICE file?

I suggested to John that we do not include anything in bin.NOTICE for dependencies that do not have a NOTICE file. I'm not sure if a final decision has been made about that, but that is my preference.

Reply:
Yes, I removed the dependencies that do not have a NOTICE file in my Daffodil PR's final changes as Steve suggested.  I'll  upload my suggested edits directly to this PR as well.